doc/Requirements.md

```diff
@@ -0,0 +1,526 @@
+(Why: The reference implementation supports it.)
+
+##### Bit field following the C abi.
+PF SHOULD support 1 or more bit sized integers.
+(Why: The reference implementation supports it.)
+
+Such parameters called are regrouped in a so called bit parameter block.
+The API to access a bit parameter block SHOULD give access to a packed bit field.
+(Why: The reference implementation supports it.)
+
+### Parameter adaptation
+
+#### Philosophy
+Parameters expose by hardware sometimes need to transform a parameter
+value. For example an hardware integer parameter could have a range 64-128 but
+it might be necessary for upper layer to access in a range 0-100.
+
+This transformation (called parameter adaptation in the rest of the document)
+could be done by the syncer. Nevertheless syncer are supposed to contain only
+business logic and should not be impacted by upper layer needs.
+
+#### Definition
+Parameter adaptation is a bijective pure function converting a parameter
+value between the syncer and other parameter reader/writer (including the inference engine).
+
+#### Requirements
+The following parameter adaptation SHOULD be supported
+ - Affine adaptation:    `affAd(value) = slope * value + offset` where slope and offset and constant coefficient
+(Why: The reference application supports it.)
+ - Logarithm adaptation: `logAd(value) = log(value)` where log() is the binary logarithm
+(Why: The reference application supports it.)
+
+A PF MAY offer Parameter adaptation composition. Ie combine multiple parameter adaptation
+Eg: composing the affine and logarithm adaptation to `compAd(value) = slope * log(value) + offset`
+(Note: The reference application supports in a tricky way: the logarithm adaptation is always combined with the affine adaptation)
+(Why: To avoid combination explosion of parameter adaptations.
+      The idea is to builtin basic function and let the user compose them to meet it's need.)
+
+## Identifiers
+Every parameter MUST have a identifier that uniquely identifies it.
+(Why: to identify a parameter outside the framework)
+
+This identifier SHOULD be a string.
+(Why: so that the (human) user can identify a parameter with ease.)
+
+Two PF instances with the same parameters MUST have the same identifier for those parameters.
+Ie. This identifier should be the same across all instances with the same configuration.
+(Why: Persistence of parameter identifier across PF instances with the same configuration.
+      To identify parameters independently of the host machine and PF instance)
+
+### Parameter tree
+A parameter SHOULD be structured in a tree. Each parameter being a distinct tree leaf.
+(Why: Tree is a simple data structure that can be easily represented and is enough to map
+      underlined layers.
+      It is not a MUST because an other structure could be chose like a list or a graph.
+      It would not change the rest of the PF properties thank to the unique identifier requirement.)
```

Comment:
A non-forest (reminder: a tree is a trivial forest) graph would break the unique identifier requirement. And a forest graph can be reduced to a tree using an additional root node.

A list is also a tree but it could arguably make sense for very simple implementations.